exposed-jdbc/src/main/kotlin/org/jetbrains/exposed/sql/statements/jdbc/JdbcDatabaseMetadataImpl.kt

```diff
@@ -112,25 +112,44 @@ class JdbcDatabaseMetadataImpl(database: String, val metadata: DatabaseMetaData)
         return schemas.map { identifierManager.inProperCase(it) }
     }
 
-    private fun ResultSet.extractColumns(tables: Array<out Table>, extract: (ResultSet) -> Pair<String, ColumnMetadata>): Map<Table, List<ColumnMetadata>> {
-        val mapping = tables.associateBy { it.nameInDatabaseCase() }
+    private fun ResultSet.extractColumns(tables: Array<out Table>, extract: (ResultSet) -> Pair<Pair<String, String>, ColumnMetadata>): Map<Table, List<ColumnMetadata>> {
+        val mapping = tables.associateBy {
+            if(currentDialect.supportsCreateSchema && it.schema == null) {
+                "$currentScheme.${it.nameInDatabaseCase()}"
+            } else {
+                // if schema is not null nameInDatabaseCase() already returns the format: schema.table
+                it.nameInDatabaseCase()
+            }
+        }
         val result = HashMap<Table, MutableList<ColumnMetadata>>()
 
         while (next()) {
-            val (tableName, columnMetadata) = extract(this)
+            val (table, columnMetadata) = extract(this)
+            val (schemaName, tableName) = table
+
             mapping[tableName]?.let { t ->
                 result.getOrPut(t) { arrayListOf() } += columnMetadata
             }
+
+            mapping["$schemaName.$tableName"]?.let { t ->
+                result.getOrPut(t) { arrayListOf() } += columnMetadata
+            }
         }
         return result
     }
 
     override fun columns(vararg tables: Table): Map<Table, List<ColumnMetadata>> {
-        val rs =  metadata.getColumns(databaseName, currentScheme, "%", "%")
+        val rs =  metadata.getColumns(null, null, "%", "%")
```

Comment:
Specifying `null` means that schema/catalog will not be used to filter the results as mentioned in the javadoc of `java.sql.DatabaseMetaData.getColumns()`:
`<code>null</code> means that the catalog name should not be used to narrow the search`

But in the implementation of Mysql5.1, if `catalog=null` the current database is used unless you've set nullCatalogMeansCurrent=false in your JDBC URL properties. see [getColumns](https://github.com/mysql/mysql-connector-j/blob/ad86f36e100e104cd926c6b81c8cab9565750116/src/com/mysql/jdbc/DatabaseMetaDataUsingInfoSchema.java#L202)

By defautlt this property is set to true see [nullCatalogMeansCurrent](https://github.com/mysql/mysql-connector-j/blob/ad86f36e100e104cd926c6b81c8cab9565750116/src/com/mysql/jdbc/ConnectionPropertiesImpl.java#L977)

For mysql8 we dont have this issue. I think here we have 3 options:
- call metadata.getColumns for every table and specify with `metadata.getColumns(table.schema, table.schema, "%", "%")`. This will make a query for every table to fetch the existing tables and columns in this schema. I recommend this but lets imagine that someone uses `createMissingTablesAndColumns` for a big number of tables, could this be **expensive** operation?
- Specify in the doc for every one using mysql5.1 to set nullCatalogMeansCurrent=false in his JDBC URL properties as it is done in this PR. see this [change](https://github.com/JetBrains/Exposed/blob/64acf90dea13f421ea906070af85d64bb6d372ab/exposed-tests/src/main/kotlin/org/jetbrains/exposed/sql/tests/DatabaseTestsBase.kt#L24) 
- Find some way to detect the version of the driver and the value of `nullCatalogMeansCurrent` property and if the version is 5.1 and nullCatalogMeansCurrent is true => throw an Exception when calling `columns(tables)`.

@Tapac what do you think?